ocaml/xapi/xapi_xenops.ml

```diff
@@ -1261,6 +1261,14 @@ module MD = struct
       else
         platformdata
     in
+    (* BIOS guests don't seem to detect the attached VTPM, block them *)
+    ( match (firmware, vm.API.vM_VTPMs) with
+    | Xenops_types.Vm.Bios, _ :: _ ->
+        let message = "Booting BIOS VM with VTPMs attached" in
+        Helpers.maybe_raise_vtpm_uninmplemented __FUNCTION__ message
```

Comment:
Should this include a reference to the VM?

Reply:
I don't think it's needed, the whole action to start a VM fails, this already signals the VM that fails to the user, both in CLI and GUI